plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -95,82 +107,157 @@ public ConnectorTableMetadata getTableMetadata()
+                public List<Object> next()
+                {
+                    return getRecord(planFilesIterator.next().file());
+                }
+
+                @Override
+                public void close()
+                        throws IOException
+                {
+                    PlanFilesIterable.super.close();
+                    closed = true;
+                }
+            };
+        }
+
+        private List<Object> getRecord(DataFile dataFile)
+        {
+            List<Object> columns = new ArrayList<>();
+            columns.add(dataFile.content().id());
+            columns.add(dataFile.path().toString());
+            columns.add(dataFile.format().name());
+            columns.add(dataFile.recordCount());
+            columns.add(dataFile.fileSizeInBytes());
+            columns.add(getIntegerBigintMapBlock(dataFile.columnSizes()));
+            columns.add(getIntegerBigintMapBlock(dataFile.valueCounts()));
+            columns.add(getIntegerBigintMapBlock(dataFile.nullValueCounts()));
+            columns.add(getIntegerBigintMapBlock(dataFile.nanValueCounts()));
+            columns.add(getIntegerVarcharMapBlock(dataFile.lowerBounds()));
+            columns.add(getIntegerVarcharMapBlock(dataFile.upperBounds()));
+            columns.add(dataFile.keyMetadata());
+            columns.add(dataFile.splitOffsets());
+            columns.add(dataFile.equalityFieldIds());
+            checkArgument(columns.size() == types.size(), "Expected %s types in row, but got %s values", types.size(), columns.size());
+            return columns;
+        }
+
+        private Block getIntegerBigintMapBlock(Map<Integer, Long> value)
+        {
+            if (value == null) {
+                return null;
             }
-            if (checkNonNull(dataFile.equalityFieldIds(), pagesBuilder)) {
-                pagesBuilder.appendIntegerArray(dataFile.equalityFieldIds());
+            return toIntegerBigintMapBlock(value);
+        }
+
+        private Block getIntegerVarcharMapBlock(Map<Integer, ByteBuffer> value)
+        {
+            if (value == null) {
+                return null;
             }
-            pagesBuilder.endRow();
-        });
+            return toIntegerVarcharMapBlock(
+                    value.entrySet().stream()
+                            .filter(entry -> idToTypeMapping.containsKey(entry.getKey()))
+                            .collect(toImmutableMap(
+                                    Map.Entry<Integer, ByteBuffer>::getKey,
+                                    entry -> Transforms.identity().toHumanString(
+                                            idToTypeMapping.get(entry.getKey()), Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+        }
 
-        return pagesBuilder.build();
-    }
+        private Block toIntegerBigintMapBlock(Map<Integer, Long> values)
+        {
+            MapType mapType = new MapType(INTEGER, VARCHAR, typeOperators);
```

Comment:
Did you mean `new MapType(INTEGER, BIGINT, typeOperators);` ?

Reply:
Yes, I am making the change and addressing it as @ebyhr suggested.